### ISSUE TITLE
When using getViteConfig, the files listed in setupFiles are loaded twice.

### ISSUE THREAD

**Issue by koyopro** [withastro/astro#12287]
### Astro Info

```block
Astro                    v4.16.7
Node                     v22.2.0
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### Describe the Bug

When using getViteConfig instead of defineConfig for Vitest, the files listed in setupFiles are loaded twice.

In the Reproducible Example, when you run `npm run test`, you can see that `console.log` is output twice as shown below.

```
❯ npm run test

stdout | tests/sample.test.ts
setup.ts imported
setup.ts imported

stdout | tests/sample.test.ts
beforeAll
beforeAll
```

I am having trouble when there is initialization processing that I want to execute only once before running the tests.

### What's the expected result?

Even when using getViteConfig, I want the files listed in setupFiles to be loaded only once.

```
❯ npm run test

stdout | tests/sample.test.ts
setup.ts imported

stdout | tests/sample.test.ts
beforeAll
```

### Link to Minimal Reproducible Example

https://stackblitz.com/github/koyopro/astro-debug/tree/vitest_setup_files?file=tests%2Fsetup.ts

### Participation

- [x] I am willing to submit a pull request for this issue.

**Comment by koyopro** [comment]
It seems that there is an issue with a plugin named `@astro/plugin-middleware`.

It looks like the problem can be resolved by modifying the following part.

https://github.com/withastro/astro/blob/9d6bcdb88fcb9df0c5c70e2b591bcf962ce55f63/packages/astro/src/core/middleware/vite-plugin.ts#L23-L27
```diff
  config(opts, { command }) {
    isCommandBuild = command === 'build';
-     return opts;
+     return {};
  },
```

https://vite.dev/guide/api-plugin.html#config
> It can return a partial config object that will be deeply merged into existing config, or directly mutate the config (if the default merging cannot achieve the desired result).

In the config, it is recommended to return only the parts you want to change, rather than the entire modified config.
Since `@astro/plugin-middleware` does not need to change the config itself, it seems correct to modify it to return an empty object as shown above.

**Comment by koyopro** [comment]
I'll try to create a PR for the fixes.

**Comment by bluwy** [comment]
Good catch. Yeah I think we can skip returning anything altogether.

**Comment by koyopro** [comment]
@bluwy Thank you for your comment!
I have submitted a Pull Request, could you please review it?
https://github.com/withastro/astro/pull/12312
